### PR TITLE
[plugin-visual] Add possibility to override baseline repository

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: docs/modules/plugins/pages/plugin-visual.adoc [hunk 1]
```diff
@@ -279,15 +284,34 @@ Establishes baseline or compares against existing one.
...
+Establishes baseline or compares against existing one using xref:developer-guides:baseline-repositories.adoc[baseline repository].
+
+[source,gherkin]
+----
+When I $actionType baseline with name `$name` using repository `$repository`
+----
+
+* `actionType` - The {visual-testing-actions}.
+* `name` - The name of baseline.
+* `repository` - The name of repository.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-visual.adoc:307]: we need to indicate that only `filesystem` is available out of the box

Comment (valfirst) [reply docs/modules/plugins/pages/plugin-visual.adoc:307]: here and in similar parameters descriptions

File: vividus-plugin-visual/src/main/java/org/vividus/visual/engine/BaselineRepositoryFactory.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.visual.engine;
+
+public interface BaselineRepositoryFactory
```
Comment (valfirst) [inline vividus-plugin-visual/src/main/java/org/vividus/visual/engine/BaselineRepositoryFactory.java:19]: why is it needed?
why not to inject just `Map<String, BaselineRepository> baselineRepositories`

**Review by valfirst — CHANGES_REQUESTED** [review]

File: docs/modules/plugins/pages/plugin-visual.adoc [hunk 3]
```diff
@@ -322,9 +373,9 @@ Establishes baseline or compares against existing one using the ignoring option.
 When I $actionType baseline with name `$name` ignoring:$ignoredElements
 ----
 
-* `actionType` - {visual-testing-actions}
-* `name` - name of baseline
-* `checkSettings` - examples table of `ELEMENT`, `AREA`, `ACCEPTABLE_DIFF_PERCENTAGE` or `REQUIRED_DIFF_PERCENTAGE`
+* `actionType` - {visual-testing-actions}.
```
Comment (uarlouski) [inline docs/modules/plugins/pages/plugin-visual.adoc:376]: can you please align formats of param names by prepending them with $?

**Review by uarlouski — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Review by valfirst — CHANGES_REQUESTED** [review]
System tests are failed

**Review by valfirst — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**